### PR TITLE
Feature/staking agents eth

### PR DESCRIPTION
**Pull request by bulbozaur**
```
Voting 20/06/2023 part 2

1) Stake all Treasury ETH in Lido
```

### CODE REVIEW

File: tests/test_2023_06_20_2.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    bypass_events_decoding,
+    vote_ids_from_env,
+    accounts,
+    interface,
+):
+    agent = interface.Agent(AGENT)
+    stETH_token = interface.StETH(STETH)
+
+    agent_steth_balance_before = stETH_token.balanceOf(agent.address)
+    agent_eth_balance_before = agent.balance()
```
Comment (bulbozaur) [inline tests/test_2023_06_20_2.py:38]: may be we want to set this with exact value

File: scripts/vote_2023_06_20_2.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+"""
+Voting
+
+1) Stake all Lido Agents ETH
```
Comment (zuzueeka) [inline scripts/vote_2023_06_20_2.py:4]: Stake all Treasury ETH in Lido

File: scripts/vote_2023_06_20_2.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    call_script_items = [
+        agent_execute(
+            target=contracts.lido.address,
+            value=current_agent_balance,
+            data=submit_calldata
+        ),
+    ]
+
+    vote_desc_items = [
+        "1) Stake all Lido Agents ETH"
```
Comment (zuzueeka) [inline scripts/vote_2023_06_20_2.py:45]: Stake all Treasury ETH in Lido

File: tests/test_2023_06_20_2.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+     # validate vote events
+    assert count_vote_items_by_events(vote_tx, contracts.voting) == 1, "Incorrect voting items count"
+
+    display_voting_events(vote_tx)
+
+    agent_steth_balance_after = stETH_token.balanceOf(agent.address)
+    agent_eth_balance_after = agent.balance()
+
+    assert almostEqWithDiff(agent_steth_balance_after, agent_steth_balance_before + agent_eth_balance_before, STETH_ERROR_MARGIN)
```
Comment (zuzueeka) [inline tests/test_2023_06_20_2.py:59]: I think it's useful to check shares as well

**Review by zuzueeka — CHANGES_REQUESTED** [review]

File: scripts/vote_2023_06_20_2.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    """Prepare and run voting."""
+
+    current_agent_balance = contracts.agent.balance()
+
+    submit_calldata = contracts.lido.submit.encode_input(ZERO_ADDRESS)
+    print(
+            contracts.lido.address,
+            current_agent_balance,
+            'Stake all Treasury ETH in Lido'
+        )
```
Comment (kadmil) [inline scripts/vote_2023_06_20_2.py:35]: Seems like trace of a debug logs?

File: scripts/vote_2023_06_20_2.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    call_script_items = [
+        agent_execute(
```
Comment (kadmil) [inline scripts/vote_2023_06_20_2.py:37]: Why the usual "agent.forward" doesn't fit?

Comment (bulbozaur) [reply scripts/vote_2023_06_20_2.py:37]: foward has no ability to pass eth value

**Review by zuzueeka — APPROVED** [review]

**Pull request merged**